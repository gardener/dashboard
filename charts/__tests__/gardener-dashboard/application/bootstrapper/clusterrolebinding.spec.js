//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { merge } = require('lodash')

const { helm } = fixtures

const renderTemplates = helm.renderBootstrapperApplicationTemplates

describe('gardener-dashboard-terminal-bootstrapper', () => {
  describe('clusterrolebinding', () => {
    let values
    let templates

    beforeEach(() => {
      values = {
        global: {
          bootstrapper: {
            enabled: true
          }
        }
      }
      templates = [
        'clusterrolebinding'
      ]
    })

    it('should not render the template', async () => {
      values.global.bootstrapper.enabled = false
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [clusterRoleBinding] = documents
      expect(clusterRoleBinding).toBe(null)
    })

    it('should render the template with default values', async () => {
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [clusterRoleBinding] = documents
      expect(clusterRoleBinding).toMatchSnapshot()
    })

    describe('when virtual garden is enabled', () => {
      beforeEach(() => {
        merge(values.global, {
          virtualGarden: {
            enabled: true,
            terminalBootstrapperUserName: 'runtime-cluster:system:serviceaccount:garden:gardener-dashboard-terminal-bootstrapper'
          }
        })
      })

      it('should render the template', async () => {
        const documents = await renderTemplates(templates, values)
        expect(documents).toHaveLength(1)
        const [clusterRoleBinding] = documents
        expect(clusterRoleBinding).toMatchSnapshot()
      })
    })
  })
})
