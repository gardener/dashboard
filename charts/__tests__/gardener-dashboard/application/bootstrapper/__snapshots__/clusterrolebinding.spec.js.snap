// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard-terminal-bootstrapper clusterrolebinding should render the template with default values 1`] = `
Object {
  "apiVersion": "rbac.authorization.k8s.io/v1",
  "kind": "ClusterRoleBinding",
  "metadata": Object {
    "labels": Object {
      "app.kubernetes.io/component": "terminal-bootstrapper",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-application-0.1.0",
    },
    "name": "gardener.cloud:system:dashboard-terminal-bootstrapper",
  },
  "roleRef": Object {
    "apiGroup": "rbac.authorization.k8s.io",
    "kind": "ClusterRole",
    "name": "gardener.cloud:system:dashboard-terminal-bootstrapper",
  },
  "subjects": Array [
    Object {
      "kind": "ServiceAccount",
      "name": "gardener-dashboard-terminal-bootstrapper",
      "namespace": "garden",
    },
  ],
}
`;

exports[`gardener-dashboard-terminal-bootstrapper clusterrolebinding when virtual garden is enabled should render the template 1`] = `
Object {
  "apiVersion": "rbac.authorization.k8s.io/v1",
  "kind": "ClusterRoleBinding",
  "metadata": Object {
    "labels": Object {
      "app.kubernetes.io/component": "terminal-bootstrapper",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-application-0.1.0",
    },
    "name": "gardener.cloud:system:dashboard-terminal-bootstrapper",
  },
  "roleRef": Object {
    "apiGroup": "rbac.authorization.k8s.io",
    "kind": "ClusterRole",
    "name": "gardener.cloud:system:dashboard-terminal-bootstrapper",
  },
  "subjects": Array [
    Object {
      "apiGroup": "rbac.authorization.k8s.io",
      "kind": "User",
      "name": "runtime-cluster:system:serviceaccount:garden:gardener-dashboard-terminal-bootstrapper",
    },
  ],
}
`;
